### PR TITLE
Home, rules, Question page & Results

### PR DESCRIPTION
**Pull request by irajbhinde**
Hey! Please review my code & let me know if any changes are to be made

Functionalities Added -

1. Home Page with Quiz Categories
2. Rules Page
3. Quiz Question Page
4. Results Page

GIFY - 

quiz

### CODE REVIEW

File: src/components/reducer/quiz-reducer.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      return {
+        ...state,
+        quizQuestions: [],
+        categoryValue: null,
+        currentQuestion: 0,
+        score: 0,
+        questionData: [],
+      };
+    default:
+      break;
```
Comment (nikhil-sharma-nks) [inline src/components/reducer/quiz-reducer.js]: you can return either entire state or an error in default

File: src/pages/Results Page/ResultsPage.jsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  const { quizState, quizDispatch } = useQuiz();
+  const navigate = useNavigate();
+  const { questionData, score } = quizState;
+
+  return (
+    <div className="resultspage-wrapper flex_c">
+      <Nav />
+      <h1 className="final-score">Final Score : {score}</h1>
+      <div className="results-container flex_c">
+        {questionData.map((data) => {
```
Comment (nikhil-sharma-nks) [inline src/pages/Results Page/ResultsPage.jsx:17]: when rendering multiple components through map, you should add key attribute

**Review by nikhil-sharma-nks — APPROVED** [review]
Looks good to me. Good work with writing clean and easy-to-understand code.

**Review by srv-pranjal — APPROVED** [review]
Good use of react concepts.
Good UI.
Clean code.
Liked the way you simplified the logic.

**Review by Digambar1411 — APPROVED** [review]
code looks fine. and liked the validation of the answer on click of option. 
check the netlify deploy logs and fix.

**Pull request merged**